### PR TITLE
Explicit Collation Error

### PR DESCRIPTION
**Pull request by king-11**
## Description

While working on #9155 it was suggested by @frouioui that we can use better error formatting to explicitly let the user know about the error being caused due to missing collation information or it is an unsupported collation.

Earlier the case for missing collation information was being suggested as a `UnsupportedComparisonError` error but given collation support is inbound we should update the error message to explicitly tell in case of `text` values that collation information is missing using `UnsupportedCollationError`

To support this the following was done:
- Error message was updated in `UnsupportedCollationError`
- `if` check with `text` types now also handles the case for `collation.Unknown`
- Accommodate the return of `UnsupportedCollationError` in `ordered_aggregate.go` and `comparer.go` so that they can use weight string in case `UnsupportedCollationError` or `UnsupportedComparisonError` is being returned
- Update the tests to compare with newly formated string

Commits Made:
- feat: explicit error for collation with text
- fix: accommodate both support and collation error
- fix: update tests to compare with new error str

## Related Issue(s)

- #8606 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by king-11** [comment]
@frouioui I have updated the tests which were already present in the main branch. I will wait for us to merge #9155. After that, I will update the tests that will be added through it. Then we can merge this PR.

**Comment by frouioui** [comment]
Sounds good to me, I'll keep myself posted

File: go/vt/vtgate/engine/comparer.go [hunk 1]
```diff
@@ -43,7 +43,8 @@ func (c *comparer) compare(r1, r2 []sqltypes.Value) (int, error) {
 	cmp, err := evalengine.NullsafeCompare(r1[colIndex], r2[colIndex], c.collationID)
 	if err != nil {
 		_, isComparisonErr := err.(evalengine.UnsupportedComparisonError)
-		if !(isComparisonErr && c.weightString != -1) {
+		_, isCollationErr := err.(evalengine.UnsupportedCollationError)
+		if !((isComparisonErr || isCollationErr) && c.weightString != -1) {
```
Comment (frouioui) [inline go/vt/vtgate/engine/comparer.go:47]: This if condition can be simplified to:
```suggestion
		if !isComparisonErr && !isCollationErr && c.weightString == -1) {
```

File: go/vt/vtgate/engine/ordered_aggregate.go [hunk 2]
```diff
@@ -412,7 +412,8 @@ func (oa *OrderedAggregate) keysEqual(row1, row2 []sqltypes.Value, colls map[int
 		cmp, err := evalengine.NullsafeCompare(row1[key.KeyCol], row2[key.KeyCol], colls[key.KeyCol])
 		if err != nil {
 			_, isComparisonErr := err.(evalengine.UnsupportedComparisonError)
-			if !(isComparisonErr && key.WeightStringCol != -1) {
+			_, isCollationErr := err.(evalengine.UnsupportedCollationError)
+			if !((isComparisonErr || isCollationErr) && key.WeightStringCol != -1) {
```
Comment (frouioui) [inline go/vt/vtgate/engine/ordered_aggregate.go:416]: same thing here, the if condition can be simplified

**Review by frouioui — APPROVED** [review]
I left two nits, but overall it looks good to me.

**Comment by king-11** [comment]
@frouioui I have made the suggested changes.

**Review by systay — APPROVED** [review]

**Comment by king-11** [comment]
@vmg awaiting your check on this PR

**Review by vmg — APPROVED** [review]

**Pull request merged**